### PR TITLE
Implemented cursor-centered mouse wheel zoom.  

### PR DESCRIPTION
**Pull request by thehans**
Issue #1857

### CODE REVIEW

**Comment by thehans** [comment]
Just a note on how I've implemented this.

I first solved the math for orthogonal projection since this was much easier to think about.  

First get the cursor coordinates in "screen space"(where [0,0] is the view center, using [this reference](https://www.scratchapixel.com/lessons/3d-basic-rendering/ray-tracing-generating-camera-rays/generating-camera-rays) for equations).
If we think about what happens when only the camera distance is changed, any object under this point ends up moving in screen space, being scaled by an amount equivalent to the (old_distance / new_distance).  To get only the difference(delta) in screen space, we subtract one from that ratio.  Now this screen space delta is scaled to match the world space, based on the orthogonal frustum size, and the camera is simply translated to correct for that perceived movement under the cursor.

I was originally thinking that after working out the orthogonal projection, I would need a separate, more complex solution for perspective projection.  However from my testing I was pleasantly surprised that it seems to match up close enough that I don't feel like any further corrections are necessary.

File: src/QGLView.cc [hunk 1]
```diff
@@ -312,12 +312,13 @@ bool QGLView::save(const char *filename)
 
 void QGLView::wheelEvent(QWheelEvent *event)
 {
+  auto pos = event->pos();
```
Comment (kintel) [inline src/QGLView.cc]: const

File: src/QGLView.cc [hunk 2]
```diff
@@ -336,6 +337,21 @@ void QGLView::zoom(double v, bool relative)
...
+  auto old_dist = cam.zoomValue();
+  this->cam.zoom(zoom, true);
+  auto dist = cam.zoomValue();
+  auto ratio = old_dist / dist - 1.0; 
+  // screen coordinates from -1 to 1
+  auto screen_x = 2.0 * (x + 0.5) / this->cam.pixel_width - 1.0;
+  auto screen_y = 1.0 - 2.0 * (y + 0.5) / this->cam.pixel_height;
+  auto height = dist * tan(cam.fov /2 * M_PI / 180);
+  auto mx = ratio*screen_x*(aspectratio*height);
+  auto mz = ratio*screen_y*height;
```
Comment (kintel) [inline src/QGLView.cc]: You can const all the local variables in this function. Good for readability as you're promising not to change the values.

File: src/QGLView.cc (hunk 1 above)
Comment (thehans) [reply src/QGLView.cc]: Thanks for the review!  Sorry, didn't mean to ignore this so long, I went ahead and committed the added `const` to those functions.

**Pull request merged**